tooling/bin/assess-minibroker-charts.rb

```diff
@@ -0,0 +1,622 @@
+  @wsaved = File.join(archive_saved, 'index.yaml') unless @wsaved
+  @wsaved
+end
+
+def archive_saved
+  @asaved = File.join(@workdir, 'ok') unless @asaved
+  @asaved
+end
+
+def archive_orig
+  @aunpatched = File.join(@workdir, 'archive-orig.tgz') unless @aunpatched
+  @aunpatched
+end
+
+def archive_patched
+  @apatched = File.join(@workdir, 'archive-patched.tgz') unless @apatched
+  @apatched
+end
+
+def manifest
+  @manifest = File.join(appdir, "manifest.yml") unless @manifest
+  @manifest
+end
+
+def chart_index
+  @chartindex = File.join(appdir, "index.yaml") unless @chartindex
+  @chartindex
+end
+
+def archive_app
+  @aapp = File.join(appdir, chart_in_app) unless @aapp
+  @aapp
+end
+
+def domain
+  @domain, _, _ = capture "kubectl get pods -o json --namespace \"#{@namespace}\" api-group-0 | jq -r '.spec.containers[0].env[] | select(.name == \"DOMAIN\").value'" unless @domain
+  @domain
+end
+
+def tester
+  @brain = File.join(@top, "make/tests") unless @brain
+  @brain
+end
+
+def appsrc
+  @appsrc = File.join(@top, "src/scf-release/src/acceptance-tests-brain/test-resources/node-env") unless @appsrc
+  @appsrc
+end
+
+def helm_repo
+  @helmrepo = "http://#{helm_app}.#{domain}" unless @helmrepo
+  @helmrepo
+end
+
+def target
+  @target = "https://api.#{domain}" unless @target
+  @target
+end
+
+def chart_ref
+  @chartref = "#{helm_repo}/#{chart_in_app}" unless @chartref
+  @chartref
+end
+
+# ......................................................................
+# Logging, and terminal cursor control
+# First three commands snarfed from test utils and modified to suit
+# (divert output into log file)
+
+# Set global options.
+# If a block is given, the options are only active in that block.
+def set(opts={})
+    if block_given?
+        old_opts = $opts.dup
+        $opts.merge! opts
+        yield
+        $opts.merge! old_opts
+    else
+        $opts.merge! opts
+    end
+end
+
+def run(*args)
+  opts = $opts.dup
+  opts.merge! args.last if args.last.is_a? Hash
+  _, _, status = capture(*args)
+  return unless opts[:errexit]
+  unless status.success?
+    # Print an error at the failure site
+    puts "Command exited with #{status.exitstatus}".red
+    fail "Command exited with #{status.exitstatus}"
+  end
+end
+
+def capture(*args)
+  _print_command(*args)
+  args.last.delete :errexit if args.last.is_a? Hash
+  stdout, stderr, status = Open3.capture3(*args)
+  log stdout
+  log stderr.red
+  return stdout.chomp, stderr.chomp, status
+end
+
+# Internal helper: print a command line in the log.
+def _print_command(*args)
+    cmd = args.dup
+    cmd.shift if cmd.first.is_a? Hash
+    cmd.pop if cmd.last.is_a? Hash
+    log "+ #{cmd.join(" ")}".bold
+    log "\n"
+end
+
+def separator(text)
+  sepline = "____________________________________________________________ #{text} ___"
+  log "\n#{sepline.magenta}\n"
+  write text
+  message = yield
+  left text.length
+  eeol
+  write message
+end
+
+def log_start(engine,enginev,chartv)
```

Comment:
Super-nit: spaces after comma; and [write] needs no space before open paren.

[write]: https://github.com/SUSE/scf/pull/2654/files#diff-f0642a28fcc55d38e482e34b24b072b6R568

Reply:
done local